src/webgpu/api/operation/render_pipeline/pipeline_output_targets.spec.ts

```diff
@@ -1,12 +1,399 @@
+    const { expectedType, expectedComponentCount } = t.getExpectedTypeAndComponentCount(format);
+    if (componentCount < expectedComponentCount) {
+      t.skip('componentCount of pipeline output must not be fewer than that of target format');
+    }
+    const info = kTextureFormatInfo[format];
+    await t.selectDeviceOrSkipTestCase(info.feature);
+
+    // expected RGBA values
+    // extra channels are discarded
+    const result = [0, 1, 0, 1];
+
+    const renderTarget = t.device.createTexture({
+      format,
+      size: { width: 1, height: 1, depthOrArrayLayers: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.RENDER_ATTACHMENT,
+    });
+
+    const pipeline = t.device.createRenderPipeline({
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+            [[stage(vertex)]] fn main(
+              [[builtin(vertex_index)]] VertexIndex : u32
+              ) -> [[builtin(position)]] vec4<f32> {
+                var pos : array<vec2<f32>, 3> = array<vec2<f32>, 3>(
+                    vec2<f32>(-1.0, -3.0),
+                    vec2<f32>(3.0, 1.0),
+                    vec2<f32>(-1.0, 1.0));
+                return vec4<f32>(pos[VertexIndex], 0.0, 1.0);
+              }
+              `,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: t.device.createShaderModule({
+          code: t.getFragmentShaderCode(result, expectedType, componentCount),
+        }),
+        entryPoint: 'main',
+        targets: [{ format }],
+      },
+      primitive: { topology: 'triangle-list' },
+    });
+
+    const encoder = t.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          view: renderTarget.createView(),
+          storeOp: 'store',
+          loadValue: { r: 1.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+    pass.setPipeline(pipeline);
+    pass.draw(3);
+    pass.endPass();
+    t.device.queue.submit([encoder.finish()]);
+
+    t.expectSingleColor(renderTarget, format, {
+      size: [1, 1, 1],
+      exp: { R: result[0], G: result[1], B: result[2], A: result[3] },
+    });
+  });
+
+g.test('color,component_count,blend')
+  .desc(
+    `Test that blending behaves correctly when:
+- fragement output has no alpha, but the src alpha is not used for the blend operation indicated by blend factors
+- attachment format has no alpha, and the dst alpha should be assumed as 1
+
+The attachment has a load value of [1, 0, 0, 1]
+`
+  )
+  .params(u =>
+    u
+      .combine('format', ['r8unorm', 'rg8unorm', 'rgba8unorm', 'bgra8unorm'] as const)
+      .beginSubcases()
+      // result is expected values in the color attachment (extra channels are discarded)
+      // output is the fragment shader output vector
+      // 0.498 -> 0x7f, 0.502 -> 0x80
+      .combineWithParams([
+        // fragment output has no alpha
+        {
+          result: [0, 1, 0, 0],
+          output: [0],
```

Comment:
I think `result` and `output` are just expected values? so they can be "private" params by spelling them `_result` and `_output`. (or maybe only `result` should be private, didn't read enough of this test yet)